Wikipedia/Code/SessionSingleton.m

```diff
@@ -201,6 +201,14 @@ - (void)fetchFinished:(id)sender
                 [self cloneSessionCookies];
             }
             break;
+            case FETCH_FINAL_STATUS_FAILED: {
+                // If autoLogin fails the credentials need to be cleared out if they're no longer valid so the
+                // user has an indication that they're no longer logged in.
+                if (error.code == LOGIN_ERROR_PASSWORD_WRONG){
```

Comment:
Two things:

1. You must check the domain and the error code. If you get back an error from NSURLConnection that happens to have the same error code as LOGIN_ERROR_PASSWORD_WRONG, then you will inadvertently clear the user name.

2. Looks like we should clear this out for all the errors you defined above, right?  All of the errors you defined seem to suggest the username or password needs to be fixed to properly log in.